docs/administrators/policy_cookbook.md

```diff
@@ -0,0 +1,251 @@
+
+### Metadata Attribute Names
+
+Use prefixes to introduce different namespaces. Below is an example that generates metadata attribute names with a special prefix. The prefix itself isn't special, but it allows implementations to reuse variable names without creating collisions.
+
+```python
+FANCY_NAMESPACE = "fancy"; # The "fancy" namespace string.
+
+# A helper rule that returns the input string with a prefix of the fancy namespace string.
+make_fancy(*name)
+{
+    FANCY_NAMESPACE ++ "::*name";
+}
+
+generate_and_use_fancy_metadata_attribute_names
+{
+    # Generate metadata attribute names under the "fancy" namespace.
+    *avu_file_path = make_fancy("file_path"); # Creates "fancy::file_path".
+    *avu_file_size = make_fancy("file_size"); # Creates "fancy::file_size".
+
+    # Do something with the variables ...
+}
+```
+
+This can be used to generate keys for `temporaryStorage` as well. See [Sharing data across PEPs](#sharing-data-across-peps) for details about using `temporaryStorage`.
+
+### Rule Names
+
+When a group of rules are related, include a rule name prefix to indicate that. You can find several examples of this in other programming languages.
+
+Below is an example that demonstrates how this may look. The rules presented make up an imaginary Native Rule Language synchronization library. In this example, the rule name prefix, `mutex_` is used to provide signal to the person writing the policy that these rules are related. The body of the rules has been left out to not take away from the primary focus of the example.
+
+```python
+#
+# Example Synchronization Library
+#
+
+# Returns a handle to a newly allocated mutex object.
+mutex_create(*mutex_name) {}
+
+# Returns the handle of the mutex identified by *mutex_name.
+# Returns zero if the mutex does not exist.
+mutex_find(*mutex_name) {}
+
+# Tries to acquire the lock on *mutex.
+# Blocks until *mutex is acquired.
+mutex_lock(*mutex) {}
+
+# Tries to acquire the lock on *mutex.
+# Blocks until *mutex is acquired or *timeout has been exceeded.
+mutex_lock(*mutex, *timeout) {}
+
+# Tries to acquire the lock on *mutex.
+# Returns true if the lock was acquired.
+# Returns false if the lock could not be acquired.
+mutex_try_lock(*mutex) {}
+
+# Releases the lock on *mutex.
+mutex_unlock(*mutex) {}
+
+# Deallocates resources for *mutex.
+mutex_destroy(*mutex) {}
+```
+
+## Sharing data across PEPs
+
+This example demonstrates how to share information across multiple PEPs within the Native Rule Engine Plugin.
+
+### How to do it ...
+
+Let's say we have the following rules defined in `/etc/irods/core.re`.
+
+Pay close attention to the use of `temporaryStorage`. It is the bridge between PEPs within the Native Rule Engine Plugin.
+
+```python
+pep_api_data_obj_put_pre(*INSTANCE_NAME, *COMM, *DATA_OBJ_INPUT, *BYTES_BUFFER, *PORTAL_OPR_OUTPUT)
+{
+    # Capture the logical path of the PUT operation.
+    # We can use it later when writing messages to the log file.
```

Comment:
Well, this info is available in the post PEP because of the `*DATA_OBJ_INPUT`.

I'll make sure to mention that the pre PEP always fires before the post PEP. I will also include a link to the dynamic PEPs page.